galen-rainbow4j/src/main/java/com/galenframework/rainbow4j/ImageHandler.java

```diff
@@ -36,34 +38,27 @@ public ImageHandler(BufferedImage image) {
     public ImageHandler(int width, int height) {
         this.width = width;
         this.height = height;
-        this.bytes = new byte[width * height * BLOCK_SIZE];
-    }
-
-    public ImageHandler(byte[] bytes, int w, int h) {
-        this.bytes = bytes;
-        this.width = w;
-        this.height = h;
+        this.bytes = ByteBuffer.allocate(width * height * BLOCK_SIZE);
     }
 
 
-    private static byte[] readRgbModelFrom(BufferedImage image) {
+    private static ByteBuffer readRgbModelFrom(BufferedImage image) {
         int w = image.getWidth();
         int h = image.getHeight();
 
         int[] pixels = new int[w * h];
```

Comment:
But does this pull request actually solve the heap errors?
This line of code is still the weakest part since it takes array of int to get the pixels and only few code lines later it transforms it into ByteBuffer. So I guess the heap error will still be there in case the screenshot image is too big. Or the heap error you got was about something else? Do you still have the complete stacktrace?
 
Anyway, thanks for your pull request! It looks good. I am only going to perform some tests before accepting it.